src/FSharp.Stats/Integration/Differentiation.fs

```diff
@@ -22,3 +26,56 @@ module Differentiation =
         let b  = (x1 - x0 - a*t1*t1)/t1
         (2.*a*t) + b
 
+    ///correcture of the Array.iterUntili function; here no boundarys are set to the lower and upper end
+    let private iterUntiliWithoutCutOff (predicate: int -> 'T -> bool) stepSize startIdx (arr: 'T []) =
+        let rec loop  (arr: 'T []) currentIdx =
+            if currentIdx < 0 then None
+            elif currentIdx > arr.Length-1 then None
+            else                                              
+                match predicate currentIdx arr.[currentIdx] with 
+                | true -> Some currentIdx   
+                | _               -> loop arr (currentIdx+stepSize) 
+        loop arr startIdx 
+
+    ///Output is the approximation for f'(x) at x.
```

Comment:
In general, write function descriptions in the style of "returns X by Y. Calculates Y by finding Z. etc."